lib/cloudregister/registerutils.py

```diff
@@ -613,11 +623,11 @@ def get_update_server_name_from_hosts(ignore_inconsistent=False):
                         'definitions, cached update server data, and '
                         'credentials file do not match')
     servers = get_available_smt_servers()
-    hosts_content = open(HOSTSFILE_PATH).read()
+    hosts_content = open(HOSTSFILE_PATH, 'rb').read()
```

Comment:
Ditto

Reply:
Not the same. Python constructs a temporary file handle. The temporary goes out of scope at the end of the function and that will close the file.

Reply:
Ack, but garbage collection in Python happens arbitrarily and not exactly when all references are removed. Chances are that it never causes an issue but the suggestion to always use context manager guarantees that the very slight chance of a confusing error can't happen.